### ISSUE TITLE
Adapting & fixing settings 

### ISSUE THREAD

**Issue by intika** [intika/Librefox#53]
Following https://github.com/intika/Librefox/issues/34 many settings have to be defaulted to a different value while leaving the choice for the user. 

**Moonchild**

- Block third-party cookies: Can block some sites (Add it as a choice) 

- Completely disable the password manager how does this improve privacy, exactly, by forcing users to type their credentials every time? ( ... ) 

- Completely disables IPv6 support. ( ... )

- Completely disables all parts of the blocklist, including known broken gfx driver issues. This will expose users to many issues with known graphics driver problems ( ... )

- Completely disables integration with the add-ons site. (addon can still be installed its just that there is no integration - add it as a choice ?)

- Completely disables extension updates (Add it as a choice)

- Completely disables Windows jumplists, because.... ( ... )

- Completely disables pre-loading of known HSTS domains; this opens the user up to first-time-visit spoofing. HSTS preloading is harmless, blocked because it's supplied by Mozilla? ( ... )

- Completely disables OCSP, but enables OCSP stapling, which won't work with disabled OCSP. ( ... )  

- Conflicting prefs with result that at best a CRL fallback is used, and at worst no checking is performed at all and revoked certs are accepted as secure. Well done Librefox, you broke https authentication checks. ( ... )

- Not forced but default; WebGL and layers acceleration is force-enabled. This will break the browser on many more systems because of GFX issues (especially hybrid and mobile chips), especially if blocklist entries aren't checked or used. ( ... )

- Completely disables webgl2 and forces webgl minimum-capability mode. This pretty much makes webgl useless. No reason to do this, since the (already enforced) fingerprinting protection already mitigates any potential webgl leaks. Fingerprinting protection doesn't enforce minimum capability mode for a reason. ( ... )
 
- Disables clipboard events, breaking many sites that use JS to place data on the clipboard... ( ... copy button still works)

- Done with lockPref (Add it as a choice)

- Considering there are plenty of duplicate entries in there you may find it frustrating that it doesn't take unless you hunt down all copies of a setting ( ... There is no duplicate but related settings).

- IMHO it's just another example of copy-pasta of insane configurations ( ... Settings have been tested but any way)

- It's not even a rebuild, it's just reconfigured ( ... )

**Comment by elypter** [comment]
>That's it really. Decide your target market, make it clear to users who that is.

it's important to know what the audience is and what audience the browser is targeted at. and it is important to communicate the priorities. something like whether it is: security>privacy>usability>maintainability or maintainability>privacy>security>usability
this makes the project consistent and avoids confusion what later seeds conflicts in larger projects. this all said, it should also be avoided to artificially constrain the project by the prime objective but only if a compromise is unavoidable.

**Comment by wolfbeast** [comment]
@Thorin-Oakenpants I actually only lifted out the ones that are severely breaking - I didn't comment on any that would be a good choice for this product with some breakage, but as you said yourself it breaks left right and center, as well as a number of things that aren't directly exposed to the user but do break under the hood (e.g. OCSP and blocklisting). I don't think most of what is done is insane, but some things that are done are clearly wrong either out of ignorance or because things haven't really been thought through before changing the settings. Some things should simply not be forced differently because all they do is break stuff with no benefit.

I understand you want to prevent calling home as much as possible, so from my own experience and the same desire for Pale Moon, here are some things to consider to improve without adding more exposure to mozilla:
- Re-enable HSTS preloading. This list is hard-coded in the build and won't download anything from Mozilla. It does have a validity period as well so even if your builds fall behind there is no danger as the internal preload list will be discarded.
- Re-enable the blocklist to prevent known dangerous and broken combinations in the graphics and plugins sections.
- Either disable Kinto blocklist updates or host them yourself. Disabling the updates will have it fall back to the binary-supplied list which is still better than no list at all.
- Enable OCSP and OCSP stapling. Most web servers with https support stapling, so there won't be the perceived privacy issue of contacting OCSP servers in that case. Where OCSP direct requests are used, it's generally not enough data to profile anything (since it will verify a certificate only once per https session), and to be fair OCSP servers are already too loaded to have tracking added to them (often so much that OCSP server connections prove unreliable from end-user machines in a lot of cases)
- Enable IPv6. Or give the user a clear choice when installing it which protocols to enable. While IPv4 is still much more prevalent for home connections, there is a slow but steady adoption of IPv6, so Librefox should be ready to accept it when it becomes available.
- Enable the password manager, but strongly suggest a sufficiently-long master password to be used. (10 or more characters)
- Enable Windows jumplists. At most this would be a local privacy concern, which is in the ream of user choice (they can use Private Browsing to ensure local privacy if needed), and it's a long-standing OS-integration feature that many people use to varying degrees.

Other things are entirely at your discretion and a balance between functionality and privacy. It'll be up to you to determine where you want Librefox to stand in that case.

**Comment by Thorin-Oakenpants** [comment]
@wolfbeast 

> I understand you want to prevent calling home as much as possible

since the post was @ me, I just want to clarify that this is not what I or the ghacks user.js aims for. If it doesn't impact security or privacy then I leave it alone (but may provide options). Some people don't understand what privacy actually means

**Comment by Thorin-Oakenpants** [comment]
> Enable Windows jumplists

you (librefox) need to distinguish between different threat levels. There's shoulder surfers vs persistent local storage (you would need to be compromised locally) vs external etc

**Comment by Thorin-Oakenpants** [comment]
an example
> Enable IPv6

After 3.5 years of NOT disabling ip6 .. I did (in the ghacks user.js). I did this after user feedback. I also including setup tags, etc ... and we weighed up the pros and cons - it doesn't break anything AFAIK, and most people worldwide can;t even use it ... and it something that should be handled on a network level ...

but it is a privacy/tracking risk, and it can compromise VPNs (when not set up correctly), etyc .. but that information is in our single point of reference .. the user.js itself .. knowledge is power

so we flipped it, after *really** thinking about it, after consulting users, and built in fallbacks for end users

^^ this is what the llibrefox project faces. Its not easy throwing 200 or 300 changes to prefs (a lot are enforcing defaults) ... I have about 90 odd (out of 450+ in FF60+) that break something - whether that be UI behavior or web sites or performance ... and the permutations of 90 items is astronomical .. which is why its so important to outlay expectations, provide info and an easy way to get to it and decipher it, etc

I'll stop now .. time for something else

**Comment by elypter** [comment]
so one question that has to be answered is "what priority does local privacy have?"  imo it's on a whole other level. once an intruder already has control over the system he can do basically anything. there are endless threat models and every counter measure can theoretically be reversed by an adversary. once an attacker is in the system there is no such thing as real scurity. al you can do is make low hanging fruits less low hanging. protecting against this is a sisyphus task. so i personally think it should have lower priority and it's mostly other softwares job anyway.
